projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts

```diff
@@ -1210,12 +1210,32 @@ export class GraphComponent implements OnInit, OnChanges, OnDestroy, AfterViewIn
     if (points.length % 2 === 1) {
       edge.midPoint = points[Math.floor(points.length / 2)];
     } else {
-      const _first = points[points.length / 2];
-      const _second = points[points.length / 2 - 1];
-      edge.midPoint = {
-        x: (_first.x + _second.x) / 2,
-        y: (_first.y + _second.y) / 2
-      };
+      let _firstX = null;
+      let _secondX = null;
+      let _firstY = null;
+      let _secondY = null;
+      const dirRight = (this.layout as Layout).settings?.properties['elk.direction'] === 'RIGHT';
```

Comment:
The layout won't always be Elk

Reply:
Please also change the name of this variable to `orientation` and make the direction more explicit.

Reply:
Done, thanks

> Please also change the name of this variable to `orientation` and make the direction more explicit.

Done, thanks